cli/flox-rust-sdk/src/providers/publish.rs

```diff
@@ -137,35 +139,33 @@ where
     }
 }
 
-fn check_builder_metadata(
-    _flox: &Flox,
-    _builder: impl ManifestBuilder,
-    _pkg: &str,
-) -> Result<CheckedBuildMetadata, String> {
-    // Build (if needed) and collect the meta data needed to publish
-    // builder.build(tmp_dir, env.path(), pkg).unwrap();
-
-    // Access a `flox build` builder
-    // ... to clone into a sandbox and run
-    // ... `flox activate; flox build;`
-    // prepublish_build();
-    // _sandbox = temp dir
-    // Use GitCommandProvider to get remote and current rev of build_repo
-    // Use GitCommandProvider to clone that remote/rev to _sandbox
-    //   this will ensure it's in the remote
-    // Load the manifest from the .flox of that repo to get the version/description/catalog
-    // Confirm access to remote resources, login, etc, so as to not waste
-    //   time if it's going to fail later or require user interaction to
-    // continue the operation.
-
-    // Obtains info from the build process (the sandboxed one above)
-    // ... base_catalog_{locked_url, rev, rev_count, rev_date}
-    // ... system, drv_path (??), outputs (??)
-    // gather_build_info(_sandbox: &Path) {
-    // Access lockfile of _sandbox to get base_catalog_{locked_url, rev, rev_count, rev_date}
-    // populate the following:
-    //      how do we get the drv_path, outputs(and paths), system from this?
-    todo!()
+/// Collect metadata needed for publishing that is obtained from the build output
+fn check_build_metadata(
+    env: &PathEnvironment,
+    pkg: &str,
+) -> Result<CheckedBuildMetadata, PublishError> {
+    // For now assume the build is successful, and present.
+    // Look for the output from the build at `results-<pkgname>`
+    // Note that the current builds only support a single output at that
+    // pre-defined path.  Later work will get structured results from the build
+    // process to feed this.
+    // See tests in build.rs for examples
+
+    let result_dir = env
+        .parent_path()
+        .map_err(|e| PublishError::UnsupportEnvironmentState(Box::new(e)))?
+        .join(format!("result-{pkg}"));
+    let store_dir = result_dir
+        .read_link()
+        .map_err(|e| PublishError::UnknownOutputs(Box::new(e)))?;
+
+    Ok(CheckedBuildMetadata {
+        outputs: vec![api_types::Output {
+            name: "bin".to_string(),
```

Comment:
nit/followup note: I think it might better follow nixpkgs convention to default to `out`

Reply:
When we get to the point of parsing the output returned by `nix build --json` then the output name will be authoritatively defined in the json response, which looks something like this:
```
flox [limeytexan/default] brantley@Michaels-MBP tmp.v7oEpQTrSg % nix build --file default.nix --json | jq .
[
  {
    "drvPath": "/nix/store/x2bdihdnp888yc13vwy154da0s8n2lqg-pkgname.drv",
    "outputs": {
      "out": "/nix/store/6rcib0i3i4c3qj1by6n4zhvbknj8486x-pkgname"
    }
  }
]
```